### DISCUSSION TITLE
Pocketbase just stopped working after some time, I restarted it and it just errors now.

### DISCUSSION THREAD

**Discussion by altacountbabi** [pocketbase/pocketbase#2125] [Q&A] [answered]
Error log:
```
panic: ReferenceError: global is not defined at node_modules/process/index.js:2:18(2)

goroutine 1 [running]:
github.com/dop251/goja_nodejs/require.Require(0xc0004d2770?, {0x17b13bc, 0x7})
        /home/runner/go/pkg/mod/github.com/dop251/goja_nodejs@v0.0.0-20221009164102-3aa5028e57f6/require/module.go:205 +0x125
github.com/dop251/goja_nodejs/process.Enable(0xc0002ad5c0?)
        /home/runner/go/pkg/mod/github.com/dop251/goja_nodejs@v0.0.0-20221009164102-3aa5028e57f6/process/module.go:30 +0x2a
github.com/pocketbase/pocketbase/plugins/jsvm.RegisterMigrations({0x259bc60?, 0xc0002ad240?}, 0x1?)
        /home/runner/work/pocketbase/pocketbase/plugins/jsvm/migrations.go:73 +0x252
github.com/pocketbase/pocketbase/plugins/jsvm.MustRegisterMigrations(...)
        /home/runner/work/pocketbase/pocketbase/plugins/jsvm/migrations.go:40
main.main()
        /home/runner/work/pocketbase/pocketbase/examples/base/main.go:71 +0x285
server@aspire-server:~/test$ ./pocketbase serve
panic: ReferenceError: global is not defined at node_modules/process/index.js:2:18(2)

goroutine 1 [running]:
github.com/dop251/goja_nodejs/require.Require(0xc00032a7e0?, {0x17b13bc, 0x7})
        /home/runner/go/pkg/mod/github.com/dop251/goja_nodejs@v0.0.0-20221009164102-3aa5028e57f6/require/module.go:205 +0x125
github.com/dop251/goja_nodejs/process.Enable(0xc0002bb5c0?)
        /home/runner/go/pkg/mod/github.com/dop251/goja_nodejs@v0.0.0-20221009164102-3aa5028e57f6/process/module.go:30 +0x2a
github.com/pocketbase/pocketbase/plugins/jsvm.RegisterMigrations({0x259bc60?, 0xc0002bb240?}, 0x1?)
        /home/runner/work/pocketbase/pocketbase/plugins/jsvm/migrations.go:73 +0x252
github.com/pocketbase/pocketbase/plugins/jsvm.MustRegisterMigrations(...)
        /home/runner/work/pocketbase/pocketbase/plugins/jsvm/migrations.go:40
main.main()
        /home/runner/work/pocketbase/pocketbase/examples/base/main.go:71 +0x285
```
This happens with any pocketbase command
Im on ubuntu 22.04 with nodejs v19

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure how to reproduce it. Do you have any custom js migrations?

**Reply by altacountbabi** [reply]
No, I just installed it today

**Reply by ganigeorgiev** [reply]
Are you sure that you don't have any custom JS file in the `pb_migrations` dir?

There are not enough details to understand and reproduce the issue. Are you able to reproduce it locally? It could be also some files permission issue similar to https://github.com/pocketbase/pocketbase/discussions/2115.

**Reply by altacountbabi** [reply]
There re some js files but I didn't create them and they all seem auto generated, and I cant reproduce this locally it works fine on my local machine (windows 11)

**Reply by ganigeorgiev** [reply]
Based on the error, it seems that the `process` goja module is conflicting with some other name matching native JS module. Do you have `node_modules` dir in the same location as the executable?

**Reply by altacountbabi** [reply]
Yes, I do

**Reply by altacountbabi** [reply]
> Yes, I do

Seems like moving it up a folder worked, thank you.

**Reply by ganigeorgiev** [reply]
Yes it is better to be in a different location or wrapped in its own folder.

Based on how [goja resolves the internal `require` statements](https://github.com/dop251/goja_nodejs/blob/2550c7b6c124b72e5a27cf765292038f43afe8bf/require/resolve.go#L14) it is possible that a conflict could occur, although I'm not able to reproduce it locally but I guess this will depend on what node modules are installed.

I may consider specifying an [explicit registry and global dir(s)](https://github.com/dop251/goja_nodejs/blob/2550c7b6c124b72e5a27cf765292038f43afe8bf/require/module.go#L78-L90) in the future to avoid eventual conflicts.

**Reply by ganigeorgiev** [reply]
@DjMeska Make sure that you don't have a `node_modules` that could conflict with the js migrations loader. If you are still not able to resolve the issue, please try to create a **minimal** reproducible repo and I'll try to investigate it.

**Reply by DjMeska** [reply]
> @DjMeska Make sure that you don't have a `node_modules` that could conflict with the js migrations loader. If you are still not able to resolve the issue, please try to create a **minimal** reproducible repo and I'll try to investigate it.

Moved this to open issue, will create public repo will see if it occurs
https://github.com/pocketbase/pocketbase/issues/2292
